### DISCUSSION TITLE
Can't use Composer

### DISCUSSION THREAD

**Discussion by davham** [Homebrew/discussions#1913] [Getting started]
I've installed Homebrew , then git, php and composer.
When I tried to run the command 
                  composer global require laravel/valet 
I get 
     illegal hardware instruction      composer global require laravel/valet .
Any Composer command will get the illegal hardware instruction.

If I uninstall homebrew, then install composer it works fine and that command works also. If I try a valet command I get a message that says I need Homebrew

I'm running 
  Mid 2010 macbook Pro (Intel Core 2 Duo)
macOS Catalina (used DosDude1 patcher to get there)

**Comment by carlocab** [comment]
>Mid 2010 macbook Pro (Intel Core 2 Duo)

This is likely your problem. Composer is trying to run hardware instructions that don't work on Core 2 Duo. It might be that Composer doesn't support your hardware at all. It's also possible that trying to patch Catalina to make it work on your hardware has misled Composer into thinking your hardware is newer than it is.

**Reply by davham** [reply]
if Homebrew in uninstalled, Composer does seem to work when it's installed manually

**Reply by SMillerDev** [reply]
It's not actually composer that's the issue here. Any php interaction through homebrews version would fail because it would be compiled for a newer CPU than you have.
